### PR TITLE
Modified Update-ModuleManifest so that params will not reset to default values

### PR DESCRIPTION
**Pull request by alerickson**
Parameters modified:  
- CompanyName
- FunctionsToExport
- CmdletsToExport 
- AliasesToExport  
- VariablesToExport 
- DscResourcesToExport

### CODE REVIEW

File: Tests/PSGetUpdateModuleManifest.Tests.ps1 [hunk 1]
```diff
@@ -113,6 +113,31 @@ Describe PowerShell.PSGet.UpdateModuleManifest -Tags 'BVT','InnerLoop' {
...
 
+    # Purpose: Validate Update-ModuleManifest will not reset original parameter values to default values
+    #
+    # Action:
+    #      Update-ModuleManifest -Path [Path] 
+    #
+    # Expected Result: The updated manifest should have the same proerty values.
+    #
+    It UpdateModuleManifestWithNoAdditionalParameters2 {    
+        New-ModuleManifest -Path $script:testManifestPath -ModuleVersion '1.0' -FunctionsToExport '*' -CmdletsToExport '*' -AliasesToExport '*' -VariablesToExport '*' -DscResourcesToExport '*'
```
Comment (bmanikm) [inline Tests/PSGetUpdateModuleManifest.Tests.ps1]: >-DscResourcesToExport '*' [](start = 167, length = 25)

This test failed on PS 4 as -DscResourcesToExport is not supported on PS 4.0 and 3.0, so conditionally add this param here.

File: Tests/PSGetUpdateModuleManifest.Tests.ps1 [hunk 2]
```diff
@@ -113,6 +113,31 @@ Describe PowerShell.PSGet.UpdateModuleManifest -Tags 'BVT','InnerLoop' {
...
+
+        #Edit company name from 'Unknown' to ''
+        (get-content $script:testManifestPath) | foreach-object {$_ -replace 'Unknown', ''} | set-content $script:testManifestPath
+        $editedModuleInfo = Test-ModuleManifest -Path $script:testManifestPath
+
+        Update-ModuleManifest -path $script:testManifestPath -ModuleVersion '2.0'
+        #$updatedModuleInfo = Test-ModuleManifest -Path $script:testManifestPath
```
Comment (bmanikm) [inline Tests/PSGetUpdateModuleManifest.Tests.ps1]: >#$updatedModuleInfo = Test-ModuleManifest -Path $script:testManifestPath [](start = 8, length = 72)

It looks like this line needs to be uncommented, right?

File: Tests/PSGetUpdateModuleManifest.Tests.ps1 [hunk 3]
```diff
@@ -113,6 +113,31 @@ Describe PowerShell.PSGet.UpdateModuleManifest -Tags 'BVT','InnerLoop' {
...
+
+        $expectedLength = 5
+        $text = @(Get-Content -Path $script:testManifestPath | Select-String "\*")
+
+        AssertEquals $updatedModuleInfo.CompanyName $editedModuleInfo.CompanyName "Company name should be $expectedCompanyName"
+        AssertEquals $($text.length) $expectedLength "Number of wildcards should be $expectedLength"
+    } `
+    -Skip:$($IsWindows -eq $False)
```
Comment (bmanikm) [inline Tests/PSGetUpdateModuleManifest.Tests.ps1]: Anything blocking us from enabling this test on non-Windows platforms?

Comment (alerickson) [reply Tests/PSGetUpdateModuleManifest.Tests.ps1]: There is nothing I know of that is, but other similar tests in this file had that condition, so I thought I would keep in line with those tests.  I can certainly take the condition out.

File: Tests/PSGetUpdateModuleManifest.Tests.ps1 (hunk 2 above)
Comment (alerickson) [reply Tests/PSGetUpdateModuleManifest.Tests.ps1]: Yes, thank you!

File: Tests/PSGetUpdateModuleManifest.Tests.ps1 (hunk 1 above)
Comment (alerickson) [reply Tests/PSGetUpdateModuleManifest.Tests.ps1]: Okay, thanks.  I'll add that check.

**Review by bmanikm — APPROVED** [review]

**Comment by alerickson** [comment]
Closes #182

**Comment by edyoung** [comment]
thanks!

**Pull request merged**